webgui-new/README.md

```diff
@@ -0,0 +1,76 @@
+# New Next.js (React) based webgui for CodeCompass
+
+## Prerequisites
+
+Make sure you are using at least version 18 of [Node.js](https://nodejs.org/en/).\
+You can also download [nvm](https://github.com/nvm-sh/nvm), and run `nvm install` at the root of the project to use the specified Node version (18.13.0).
+
+## Development
+
+For easier development, first run the webserver (CodeCompass_webserver), then run the development server.\
+Since the webserver and development server are different origins, you have to use the [CORS extension](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf?hl=en) to resolve CORS issues.
+
+Alternatively, you could add this change to the `MainRequestHandler::begin_request_handler(struct mg_connection* conn_)` method in `mainrequesthandler.cpp`, then build the webserver again.\
+(Note: Do not commit this change, as it is unnecessary in production.)
+
+```c
+mg_send_header(conn_, "Access-Control-Allow-Origin", "http://localhost:3000");
+```
+
+Learn more about CORS [here](https://www.moesif.com/blog/technical/cors/Authoritative-Guide-to-CORS-Cross-Origin-Resource-Sharing-for-REST-APIs/).
+
+Run the webserver with:
+
+```bash
+cd <path-to-webserver-binary> && ./CodeCompass_webserver -w <path-to-workspace-folder> -p <port>
+```
+
+Install npm packages with:
+
+```bash
+npm install
+```
+
+Generate TypeScript files from Thrift by running the following command:
+
+```bash
+chmod +x thrift-codegen.sh && ./thrift-codegen.sh --thrift-source ../
+```
+
+You can specify the properties of the webserver by creating a `.env` file at the root of the project. This is only required for development.\
+The default values are: `localhost` for host, `8080` for port, `false` for https, and path will be an empty string.\
+Note: you need to set the `WEBSERVER_PORT`, if you choose to use another port for the webserver, the other properties are not that important considering you run a local webserver.
+
+```py
+# .env
+WEBSERVER_HOST=
+WEBSERVER_PORT=
+WEBSERVER_HTTPS=
+WEBSERVER_PATH=
```

Comment:
It could be a simplification to replace this with a simple parameter, e.g. `BACKEND_URL`. Then with the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) type you could create an object from it to easily get the schema, the domain, the port and the path.